### DISCUSSION TITLE
Team Libraries - Sharing Tokens

### DISCUSSION THREAD

**Discussion by cmaker1** [six7/figma-tokens#62] [Q&A]
Hello,

Designer here. :) Your plugin is the closest one I've seen to filling the gap between the language designers speak and engineering teams when it comes to tokens. Thank you! 

As I'm digging in, I am trying to find a way where another team consuming our Team Libraries can utilize the plugin to view the tokens it is consuming from a team library. It seems that the styles need to be local to the file and not from a shared library. Am I correct about this?

I post this in ideas because it seems that while designing a Token Based Design System half of the equation is Figma Team Libraries and might be an opportunity you'd like to consider. Without being able to view the tokens via libraries I'm not sure how desirable this tool will be for larger teams. 

I hope I have missed this feature and please let me know if I have!! 

Best regards,
Clare

**Comment by six7** [comment]
Hey!

First of all glad you find it useful 🙏 

I think I can provide 2 answers:

Yes, Styles currently only work with local styles unfortunately. However I think this is something that can be fixed in the near term when we save references to the style id to each token, so you wouldnt only store the color token but also the style association.

Putting tokens to some sort of team library will be done by an external token sync. Unfortunately we cant store tokens in a good way on Figmas Team library itself (as we dont have write or read access as a plugin to anything outside of the current file) but being able to sync it for the whole team is high on the prio list 🙏 Without that its really hard to work on different files with the same tokens..

Hope I answered that correctly 👍

**Reply by cmaker1** [reply]
Hello Jan Six,

Thank you for your reply. This makes sense. I will stay tuned for updates! Best of luck in the new year!!

Take care! 
Clare

**Reply by six7** [reply]
FYI: Token sync is now enabled, letting you store your tokens on JSONBin for a start
